okhttp/src/main/java/okhttp3/HttpUrl.kt

```diff
@@ -1368,17 +1388,17 @@ class HttpUrl internal constructor(builder: Builder) {
         for (i in pos + 1 until limit) {
           val c = input[i]
 
-          if (c >= 'a' && c <= 'z'
-              || c >= 'A' && c <= 'Z'
-              || c >= '0' && c <= '9'
-              || c == '+'
-              || c == '-'
-              || c == '.') {
-            continue // Scheme character. Keep going.
+          return if (c in 'a'..'z'
+                  || c in 'A'..'Z'
+                  || c in '0'..'9'
+                  || c == '+'
+                  || c == '-'
+                  || c == '.') {
+              continue // Scheme character. Keep going.
```

Comment:
This one I think I don't agree with. If we're hoisting a `return` from an `if` I would prefer it only return or throw and not mix other control flows like `continue`.